service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageDocumentService.java

```diff
@@ -240,16 +246,21 @@
 
         C2DocumentBundle c2DocumentBundle = caseData.getC2DocumentBundleByUUID(selected);
 
-        List<Element<SupportingEvidenceBundle>> updatedCorrespondenceDocuments =
+        List<Element<SupportingEvidenceBundle>> modifiedEvidence =
             setDateTimeUploadedOnSupportingEvidence(caseData.getSupportingEvidenceDocumentsTemp(),
                 c2DocumentBundle.getSupportingEvidenceBundle());
 
-        List<Element<C2DocumentBundle>> updatedC2Bundles = new ArrayList<>();
-        for (Element<C2DocumentBundle> c2DocumentBundleElement : caseData.getC2DocumentBundle()) {
-            if (selected.equals(c2DocumentBundleElement.getId())) {
-                c2DocumentBundleElement.getValue().setSupportingEvidenceBundle(updatedCorrespondenceDocuments);
+        List<Element<C2DocumentBundle>> updatedC2Bundles = caseData.getC2DocumentBundle();
```

Comment:
vv minor: don't like the name updated here because it is a while in reading terms until it is actually updated

Reply:
changed to c2Bundles